src/main/java/org/jitsi/jigasi/transcription/OracleTranscriptionService.java

```diff
@@ -0,0 +1,353 @@
+                    .isAckEnabled(false)
+                    .finalSilenceThresholdInMs(1000)
+                    .build();
+            try
+            {
+                client.open(websocketUrl, 443, realtimeClientParameters);
+                sessionStart = Instant.now();
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while connecting to OCI service", e);
+            }
+        }
+
+        @Override
+        public void sendRequest(TranscriptionRequest request)
+        {
+            connect();
+            try
+            {
+                client.sendAudioData(request.getAudio());
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while sending audio data to OCI service", e);
+            }
+        }
+
+        @Override
+        public void end()
+        {
+            logger.info("Ending OCI session and waiting for the last results");
+            try
+            {
+                // Wait for 5 seconds for the last results to come in
+                CountDownLatch latch = new CountDownLatch(1);
+                latch.await(5, TimeUnit.SECONDS);
+            }
+            catch (InterruptedException e)
+            {
+                logger.warn("Interrupted while waiting for the last results", e);
+            }
+
+            try
+            {
+                client.close();
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while closing the OCI connection", e);
+            }
+            sessionEnding = true;
+        }
+
+        @Override
+        public boolean ended()
+        {
+            return sessionEnding;
+        }
+
+        @Override
+        public void addTranscriptionListener(TranscriptionListener listener)
+        {
+            listeners.add(listener);
+        }
+
+        @Override
+        public void onClose(int statusCode, String statusMessage)
+        {
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Connection closed: " + statusCode + " " + statusMessage);
+            }
+        }
+
+        @Override
+        public void onAckMessage(RealtimeMessageAckAudio ackMessage)
+        {
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Received ack message: " + ackMessage);
+            }
+        }
+
+        @Override
+        public void onResult(RealtimeMessageResult result)
+        {
+            if (!result.getTranscriptions().isEmpty())
+            {
+                String tsResult = result.getTranscriptions().get(0).getTranscription().trim();
+                Boolean isFinal = result.getTranscriptions().get(0).getIsFinal();
+
+                for (TranscriptionListener l : listeners)
+                {
+                    l.notify(new TranscriptionResult(
+                            null,
```

Comment:
Is this session not associated with a specific `Participant`?

Reply:
no, check the Google and Whisper transcribers